build.sh

```diff
@@ -0,0 +1,178 @@
+  git fetch --all || _die "Git fetch failed for Playwright"
+  git checkout "${PLAYWRIGHT_VERSION}" || _die "Can't checkout playwright version ${PLAYWRIGHT_VERSION}.."
+  popd > /dev/null || exit 1
+}
+
+_checkout_playwright() {
+  _status "Cloning playwright into ${PLAYWRIGHT_DIR}"
+  git clone https://github.com/microsoft/playwright.git "${PLAYWRIGHT_DIR}" || _die "Cloning playwright failed!"
+}
+
+_check_repo_foxhound_repo_state() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+ test -z "$(git status --porcelain)" || _die "Dirty Worktree, please commit all changes before running the build script!"
+  popd > /dev/null || exit 1
+}
+
+_prepare_foxhound() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+ _status "Preparing the foxhound build environment"
+  if [ -d "./juggler" ]; then
+    _status "Deleting stale juggler"
+    rm -rf "${FOXHOUND_DIR}/juggler"
+  fi
+  if [ ! -f "${FOXHOUND_DIR}/.mozconfig" ]; then
+    _status "Setting default mozconfig from Ubuntu profile"
+    cp "${FOXHOUND_DIR}/taintfox_mozconfig_ubuntu" "${FOXHOUND_DIR}/.mozconfig" 
+  fi
+  if [ -n "$NO_CLOBBER" ]; then
+    ./mach --no-interactive clobber
+  fi
+ ./mach --no-interactive bootstrap --no-system-changes --application-choice=browser
+  popd > /dev/null || exit 1
+}
+
+_patch_foxhound() {
+  _status "Patching foxhound for playwright integration"
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+  cp -r "${PLAYWRIGHT_DIR}/browser_patches/firefox/juggler" "juggler"
+  git apply --verbose --index --whitespace=nowarn --recount "${PLAYWRIGHT_DIR}/browser_patches/firefox/patches"/* || _die "Playwright patches failed to apply."
+  popd > /dev/null || exit 1
+}
+
+_build_foxhound() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+  ./mach build || _die "./mach build error"
+  popd > /dev/null || exit 1
+}
+
+_package_foxhound() {
+  pushd "${FOXHOUND_DIR}" > /dev/null || _die "Can't change into foxhound dir: ${FOXHOUND_DIR}"
+ _status "Packaging foxhound.."
+  ./mach package || _die "./mach package error"
+
+  if [ -n "$WITH_PLAYWRIGHT_INTEGRATION" ]; then
+    mkdir -p "${FOXHOUND_OBJ_DIR}/dist/foxhound/defaults/pref"
+    cp "${PLAYWRIGHT_DIR}/browser_patches/firefox/preferences/playwright.cfg" "${FOXHOUND_OBJ_DIR}/dist/foxhound/"
+    cp "${PLAYWRIGHT_DIR}/browser_patches/firefox/preferences/00-playwright-prefs.js" "${FOXHOUND_OBJ_DIR}/dist/foxhound/defaults/pref/"
+  fi
+  pushd "${FOXHOUND_OBJ_DIR}/dist" || exit 1
+  zip -r foxhound_linux.zip foxhound
+  _status "Zip located at '$(pwd  || true)/foxhound_linux.zip', done!"
+  popd > /dev/null || exit 1
+  popd > /dev/null || exit 1
+}
+
+
+
+main() {
+  FOXHOUND_DIR="${CURRENT_DIR}"
+  PLAYWRIGHT_DIR="${BASEDIR}/playwright"
```

Comment:
Does having the playwright directory as a subdirectory cause any problems? e.g. for git? Do we need to add it to .gitignore?

Reply:
Ah, BASEDIR refers to the directory where the Foxhound checkout resides. For example, /opt/foxhound/build.sh would lead to a playwright checkout in /opt/playwright. I thought about allowing to change the playwright directory, but that makes it fairly easy to forget the option and end up with several checkouts.

Reply:
That makes sense!